tests/integration/basic.cpp

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) 2016-2020 Memgraph Ltd. [https://memgraph.com]
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include "mgclient.h"
+
+template <typename T>
+T GetEnvOrDefault(const std::string &value_name, const T &default_value) {
+  const char *char_value = std::getenv(value_name.c_str());
+  if (!char_value) return default_value;
+  T value;
+  std::stringstream env_value_stream(char_value);
+  env_value_stream >> value;
+  return value;
+}
+
+void CheckMgValueType(const mg_value *value, const mg_value_type value_type) {
+  ASSERT_EQ(mg_value_get_type(value), value_type);
+}
+
+bool GetBoolValue(const mg_value *value) {
+  CheckMgValueType(value, MG_VALUE_TYPE_BOOL);
+  return mg_value_bool(value);
+}
+
+int GetIntegerValue(const mg_value *value) {
+  CheckMgValueType(value, MG_VALUE_TYPE_INTEGER);
+  return mg_value_integer(value);
+}
+
+const mg_node *GetNodeValue(const mg_value *value) {
+  CheckMgValueType(value, MG_VALUE_TYPE_NODE);
+  return mg_value_node(value);
+}
+
+const mg_relationship *GetRelationshipValue(const mg_value *value) {
+  CheckMgValueType(value, MG_VALUE_TYPE_RELATIONSHIP);
+  return mg_value_relationship(value);
+}
+
+std::string GetStringValue(const mg_string *value) {
+  return std::string(mg_string_data(value), mg_string_size(value));
+}
+
+std::string GetStringValue(const mg_value *value) {
+  CheckMgValueType(value, MG_VALUE_TYPE_STRING);
+  const mg_string *s = mg_value_string(value);
+  return GetStringValue(s);
+}
+
+class MemgraphConnection : public ::testing::Test {
+ protected:
+  virtual void SetUp() override {
+    params = mg_session_params_make();
+    std::string memgraph_host =
+        GetEnvOrDefault<std::string>("MEMGRAPH_HOST", "127.0.0.1");
+    int memgraph_port = GetEnvOrDefault<int>("MEMGRAPH_PORT", 7687);
+    bool memgraph_ssl = GetEnvOrDefault<bool>("MEMGRAPH_SSLMODE", true);
+
+    mg_session_params_set_host(params, memgraph_host.c_str());
+    mg_session_params_set_port(params, memgraph_port);
+    mg_session_params_set_sslmode(
+        params, memgraph_ssl ? MG_SSLMODE_REQUIRE : MG_SSLMODE_DISABLE);
+  }
+
+  virtual void TearDown() override {
+    mg_result *result;
+    mg_session_run(session, "MATCH (n) DETACH DELETE n", NULL, NULL);
+    mg_session_pull(session, &result);
```

Comment:
You can use `ASSERT_*` in `TearDown`.
It is better to verify the cleanup commands in the `TearDown` function than making it necessary to copy the cleanup logic into each test.